nipype/interfaces/fsl/epi.py

```diff
@@ -1232,3 +1232,206 @@ def _run_interface(self, runtime):
         if runtime.stderr:
             self.raise_exception(runtime)
         return runtime
+
+
+class EddyQuadInputSpec(FSLCommandInputSpec):
+    base_name = traits.Str(
+        'eddy_corrected',
+        usedefault=True,
+        argstr='%s',
+        desc=("Basename (including path) for EDDY output files, i.e., "
+              "corrected images and QC files"),
+        position=0,
+    )
+    idx_file = File(
+        exists=True,
+        mandatory=True,
+        argstr="--eddyIdx=%s",
+        desc=("File containing indices for all volumes into acquisition "
+              "parameters")
+    )
+    param_file = File(
+        exists=True,
+        mandatory=True,
+        argstr="--eddyParams=%s",
+        desc="File containing acquisition parameters"
+    )
+    mask_file = File(
+        exists=True,
+        mandatory=True,
+        argstr="--mask=%s",
+        desc="Binary mask file"
+    )
+    bval_file = File(
+        exists=True,
+        mandatory=True,
+        argstr="--bvals=%s",
+        desc="b-values file"
+    )
+    bvec_file = File(
+        exists=True,
+        argstr="--bvecs=%s",
+        desc=("b-vectors file - only used when <base_name>.eddy_residuals "
+              "file is present")
+    )
+    output_dir = traits.Str(
+        name_template='%s.qc',
+        name_source=['base_name'],
+        argstr='--output-dir=%s',
+        desc="Output directory - default = '<base_name>.qc'",
+    )
+    field = File(
+        argstr='--field=%s',
+        desc="TOPUP estimated field (in Hz)",
+    )
+    slice_spec = File(
+        argstr='--slspec=%s',
+        desc="Text file specifying slice/group acquisition",
+    )
+    verbose = traits.Bool(
+        False,
+        argstr='--verbose',
+        desc="Display debug messages",
+    )
+
+
+class EddyQuadOutputSpec(TraitedSpec):
+    out_qc_json = File(
+        exists=True,
+        mandatory=True,
+        desc=("Single subject database containing quality metrics and data "
+              "info.")
+    )
+
+    out_qc_pdf = File(
+        exists=True,
+        mandatory=True,
+        desc="Single subject QC report."
+    )
+
+    out_avg_b_png = traits.List(
+        File(
+            exists=True,
+            mandatory=True,
+            desc=("Image showing mid-sagittal, -coronal and -axial slices of "
+                  "each averaged b-shell volume.")
```

Comment:
`desc` should be a parameter to `List`, not `File`, and you can drop `mandatory`. I've suggested removing it from the other output spec files... I don't think it's enforced, but if it is, it will cause problems if you have the `remove_unnecessary_outputs` config option set, and no node takes those files as inputs.